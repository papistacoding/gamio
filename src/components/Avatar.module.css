.wrapper {
  display: flex;
  flex-direction: column;
  align-items: center;
}
.avatarImage {
  margin-top: -15%;
  z-index: 1;
  width: 26%;
  border-radius: 50%;
}
.infoWrapper {
  display: flex;
  margin-top: 1rem;
}
.playerName {
  flex: 0 0 100%;
  text-align: center;
}
.playerName h2, .playerName h5 {
  text-transform: uppercase;
}
.level {
  font-family: "PrimaryBold", sans-serif;
  font-weight: 900;
  font-size: 14px;
}
@media screen and (min-width: 1025px) {
  .wrapper {
    position: absolute;
    top: 50%;
    flex-direction: row;
    z-index: 1;
    left: 6%;
    gap: 2rem;
  }
  .avatarImage {
    margin: unset;
    z-index: unset;
    width: 9.45vw;
  }
  .playerName {
    text-align: left;
    flex: 0 0 91%;
  }
  .playerName h2 {
    color: white;
  }
}
