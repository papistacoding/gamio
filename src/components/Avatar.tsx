import React, { useContext } from "react";
import classes from "./Avatar.module.css";
import { UserContext } from "@/store/UserProvider";
import { IUser } from "@/interface/user";

export const Avatar: React.FC = () => {
  const user: IUser | null = useContext(UserContext);
  console.log(user);

  return (
    <div className={classes.wrapper}>
      <img className={classes.avatarImage} src={user?.AvatarUrl} alt="Avatar" />
      <div className={classes.infoWrapper}>
        <div className={classes.playerName}>
          <h2 className="color-dark">{user?.Name}</h2>
          <h5 className="color-gray">@{user?.Username}</h5>
        </div>
        <span className={`${classes.level} color-blue`}>{user?.level}</span>
      </div>
    </div>
  );
};
