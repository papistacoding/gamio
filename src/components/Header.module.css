.header {
  position: relative;
  width: 100%;
}
.headerImage {
  width: 100%;
}
.headerDesktop {
  position: relative;
}
.clip {
  display: none;
}
.bluredMark {
  display: none;
}
@media screen and (min-width: 1025px) {
  .clip {
    display: block;
    bottom: -5%;
    width: 17.223vw;
    height: 21.39vw;
    position: absolute;
    clip-path: polygon(0 0, 50% 0, 80% 100%, 0% 100%);
    background-color: #4175fa;
  }
  .blurredMark {
    display: block;
    position: absolute;
    right: 0;
    bottom: -70%;
    background: rgba(252, 105, 71, 0.1);
    filter: blur(150px);
    width: 28%;
    height: 600px;
  }
}
