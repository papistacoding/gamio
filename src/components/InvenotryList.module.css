.inventoryList {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  gap: 2rem;
  margin: 3rem 0;
  padding: 0 2rem;
}
@media screen and (min-width: 1025px) {
 .inventoryList{
  margin: 3rem 0 3rem 7.45%;
  justify-content: unset;
  padding: 0;
  gap: 2rem;
 }
}