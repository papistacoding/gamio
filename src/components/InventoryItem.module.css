.itemWrapper {
  display: flex;
  flex-direction: column;
  width: 280px;
  height: 420px;
  box-shadow: 0px 6px 16px rgba(177, 181, 188, 0.2);
  align-items: center;
  justify-content: space-evenly;
  transition: transform 0.2s ease-in-out;
  z-index: 1;
}

.itemWrapper:nth-child(n + 3) {
  background-color: #f7f7f7;
}
.itemWrapper:nth-child(n + 4) {
  border: 2px solid;
  border-image-slice: 1;
  border-width: 2px;
  border-image-source: linear-gradient(
    to right,
    rgba(66, 118, 250, 0.4),
    rgba(252, 105, 71, 0.4)
  );
}
.imageWrapper {
  width: 78%;
  height: 52%;
  display: flex;
  justify-content: center;
}
.image {
  max-width: 100%;
  max-height: 100%;
}
.info {
  max-width: 78.55%;
  text-align: center;
}
.info > span {
  font-family: "PrimaryMeduim", sans-serif;
  font-weight: 600;
  font-size: 12px;
  padding: 4px 8px;
  border-radius: 4px;
}
.info h3 {
  font-family: "PrimaryBold", sans-serif;
  font-weight: 900;
  font-size: 18px;
  text-transform: uppercase;
}
.codeWrapper {
  display: flex;
  justify-content: space-between;
  align-items: center;
  font-family: "PrimaryMeduim", sans-serif;
  font-weight: 600;
  font-size: 12px;
  background-color: transparent;
  border: #e5e5e5 1px solid;
  color: #495967;
  cursor: pointer;
}
.codeWrapper > img {
  padding-left: 0.5rem;
}
.codeWrapper > div {
  display: flex;
  width: 65%;
  height: 26px;
  align-items: center;
  background-color: #eef3ff;
  padding-right: 0.5rem;
  position: relative;
}
.codeWrapper > div span {
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
  text-transform: uppercase;
  padding: 0 0.5rem;
  color: #6f819c;
}
.claimed {
  color: #85ee00;
  background: rgba(133, 238, 0, 0.1);
}
.owned {
  color: #ffffff;
  background-color: #4175fa;
}
.used {
  color: #6f819c;
  background-color: #e5e5e5;
}
.orderButton {
  font-family: "PrimaryMeduim", sans-serif;
  font-weight: 600;
  color: white;
  padding: 5px 24px;
  height: 40px;
  background: #4175fa;
  border-radius: 4px;
  box-shadow: 0px 6px 16px rgb(77 77 77 / 34%);
  border: none;
  width: 100%;
  transition: transform 0.2s ease-in-out;
  cursor: pointer;
}
.orderButton:active {
  transform: scale(0.95);
}
.tooltip {
  font-family: "PrimaryMedium", sans-serif;
  font-weight: 600;
  position: absolute;
  background-color: #4175fa;
  color: white;
  padding: 10px;
  border-radius: 5px;
  z-index: 9999;
  transition: opacity 0.2s ease-in-out;
  top: -150%;
  left: 50%;
  transform: translateX(-50%);
  text-align: center;
}

.tooltip:after {
  content: "";
  position: absolute;
  bottom: -34px;
  left: 50%;
  transform: translateX(-50%);
  border-width: 11px;
  border-style: solid;
  border-color: #4175fa transparent transparent transparent;
  z-index: 123123;
  height: 16px;
  width: 0px;
}
@media (min-width: 768px) and (max-width: 1440px) {
  .itemWrapper {
    width: 40%;
    max-width: 280px;
  }

}
@media (min-width: 1025px){
  .itemWrapper:hover {
    transform: translateY(-5px);
  }
}