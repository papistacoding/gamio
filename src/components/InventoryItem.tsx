import React, { useState } from "react";
import classes from "./InventoryItem.module.css";
import { formatDate } from "@/helpers/formatDate";
import { copyToClipboard } from "@/helpers/copyToClipboard";

interface InventoryItemProps {
  itemData: IItem;
}

interface IStatus {
  status: string;
  class: string;
}

export const InventoryItem: React.FC<InventoryItemProps> = ({ itemData }) => {
  const [showTooltip, setShowTooltip] = useState<boolean>(false);

  const renderItemStatus = (): IStatus => {
    if (itemData.type === "code") {
      if (itemData.used) {
        return { status: "USED", class: classes.used };
      } else if (itemData.claimed) {
        return { status: "CLAIMED", class: classes.claimed };
      } else {
        return { status: "UNCLAIMED", class: classes.unclaimed };
      }
    } else {
      //assuming we only have codes and hardwares
      if (itemData.shipped && itemData.shipping_date) {
        return {
          status: `SHIPPED ${formatDate(itemData.shipping_date)}`,
          class: classes.date,
        };
      } else {
        return { status: "OWNED", class: classes.owned };
      }
    }
  };

  const status: IStatus = renderItemStatus();

  const handleCodeClick = (): void => {
    copyToClipboard(itemData.code);
    setShowTooltip(true);
    setTimeout(() => {
      setShowTooltip(false);
    }, 2000);
  };

  const renderCode = (): JSX.Element => {
    return (
      <div className={classes.codeWrapper} onClick={handleCodeClick}>
        <img src="/images/code.svg" />
        <span>CODE</span>
        <div>
          {showTooltip && <div className={classes.tooltip}>Copied!</div>}
          <span>{itemData.code}</span> <img src="/images/clipboard.svg" />
        </div>
      </div>
    );
  };

  const renderButton = (): JSX.Element | null => {
    return !itemData.shipped ? (
      <button className={classes.orderButton}>
        <span>ORDER SHIPPING</span>
      </button>
    ) : null;
  };

  return (
    <div className={classes.itemWrapper}>
      <div className={classes.imageWrapper}>
        <img className={classes.image} src={itemData.image_url} alt="item image" />
      </div>
      <div className={classes.info}>
        <span className={status.class}>{status.status}</span>
        <h3 className="color-dark">{itemData.title}</h3>
        {itemData.type === "code" ? renderCode() : renderButton()}
      </div>
    </div>
  );
};
