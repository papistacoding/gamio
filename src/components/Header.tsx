import React from "react";
import { Avatar } from "./Avatar";
import classes from "./Header.module.css";
import { Navigation } from "./Navigation";

export const Header: React.FC = () => {
  return (
    <>
      <div className={classes.headerDesktop}>
        <div className={classes.header}>
          <img
            className={classes.headerImage}
            src="/images/header_img.svg"
            alt="header"
            data-testid="header-image"
          />
        </div>
        <Avatar data-testid="avatar" />
        <div className={classes.clip}></div>
        <Navigation data-testid="navigation" />
        <div className={classes.blurredMark}></div>
      </div>
    </>
  );
};
