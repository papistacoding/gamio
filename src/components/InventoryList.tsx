import React from "react";
import { InventoryItem } from "./InventoryItem";
import classes from "./InvenotryList.module.css";

interface InventoryListProps {
  data: IItem[];
}

export const InventoryList: React.FC<InventoryListProps> = ({ data }) => {
  return (
    <div className={classes.inventoryList}>
      {data && data.length ? (
        data.map((item: IItem) => <InventoryItem key={item.Id} itemData={item} />)
      ) : (
        <span>There are no items in the Inventory</span>
      )}
    </div>
  );
};