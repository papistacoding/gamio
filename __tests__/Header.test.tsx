import React from "react";
import { render } from "@testing-library/react";
import { Header } from "@/components/Header";

jest.mock("next/router", () => require("next-router-mock"));

describe("Header", () => {
  it("matches snapshot", () => {
    const { asFragment } = render(<Header />);
    expect(asFragment()).toMatchSnapshot();
  });
});
