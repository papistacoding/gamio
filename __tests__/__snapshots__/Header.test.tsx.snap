// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="headerDesktop"
  >
    <div
      class="header"
    >
      <img
        alt="header"
        class="headerImage"
        data-testid="header-image"
        src="/images/header_img.svg"
      />
    </div>
    <div
      class="wrapper"
    >
      <img
        alt="Avatar"
        class="avatarImage"
      />
      <div
        class="infoWrapper"
      >
        <div
          class="playerName"
        >
          <h2
            class="color-dark"
          />
          <h5
            class="color-gray"
          >
            @
          </h5>
        </div>
        <span
          class="level color-blue"
        />
      </div>
    </div>
    <div
      class="clip"
    />
    <div
      class="navWrapper"
    >
      <a
        class=" link   color-gray"
        href="/"
      >
        OVERVIEW
      </a>
      <a
        class=" link   color-gray"
        href="/inventory"
      >
        INVENTORY
      </a>
      <a
        class=" link   color-gray"
        href="/settings"
      >
        SETTINGS
      </a>
    </div>
    <div
      class="blurredMark"
    />
  </div>
</DocumentFragment>
`;
